streams/src/main/java/org/apache/kafka/streams/state/internals/MeteredKeyValueStore.java

```diff
@@ -103,8 +104,12 @@ public void init(final ProcessorContext context,
 
     @SuppressWarnings("unchecked")
     void initStoreSerde(final ProcessorContext context) {
+        final InternalProcessorContext internalProcessorContext = (InternalProcessorContext) context;
+        final String storeName = name();
+        final String changelogTopic = internalProcessorContext.changelogFor(storeName);
         serdes = new StateSerdes<>(
-            ProcessorStateManager.storeChangelogTopic(context.applicationId(), name()),
+            changelogTopic != null
```

Comment:
For which case would it be `null`?

Reply:
It would be `null` for the case where logging is disabled. In this case we would still need the store serdes to read records from the store. We just pass in the default changelog name, so that one can still use a changelog name in the serde, e.g., as subject name for Confluent's schema registry.